### PR TITLE
Try run json transform only once for page

### PR DESCRIPTION
**Pull request by mingwli**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5212)

### CODE REVIEW

**Comment by mingwli** [comment]
trying to modify input may influence enumeration. see #4960
close this one

**Comment by mingwli** [comment]
> See #4960 , you cannot change the input token since `JToken` is not immutable and the same reference is shared with / cached in other components.

As adviced by @yufeih, try copy clone before running schema content transform

File: src/docfx/build/page/BuildPage.cs [hunk 1]
```diff
@@ -252,22 +252,21 @@ private static async Task<(List<Error> errors, JObject model)> LoadJson(Context
             var schemaValidationErrors = schemaTemplate.JsonSchemaValidator.Validate(obj);
             errors.AddRange(schemaValidationErrors);
 
-            // transform model via json schema
-            var (schemaTransformError, transformedToken) = schemaTemplate.JsonSchemaTransformer.TransformContent(file, context, obj);
-            errors.AddRange(schemaTransformError);
-            var pageModel = (JObject)transformedToken;
+            obj = (JObject)JsonUtility.ShallowClone(obj);
```
Comment (yufeih) [inline src/docfx/build/page/BuildPage.cs]: nit: consider give it a different name if the variable is mutated:

```suggestion
            var xxx = (JObject)JsonUtility.ShallowClone(obj);
```

Comment (mingwli) [reply src/docfx/build/page/BuildPage.cs]: using JsonUtility.Merge for shallow clone

**Comment by yufeih** [comment]
The diff shows that FileLinkBuilder does not de-dup results

**Pull request merged**